### DISCUSSION TITLE
Open modal from url queryString

### DISCUSSION THREAD

**Discussion by slzrdev** [livewire/livewire#2831] [1. Help]
I'm trying to be able to share a link including a modal with the product defined by the url. I'm using the modal provided by Jetstream with some modifications.

I'm using livewire 2, laravel 8. AlpineJs 2.8.

I've updated this question to make it clearer. 

I call the modal within the blade view and everything works well, my problem is that I can't share the link to a specific product. This is the component:

    namespace App\Http\Livewire;
    use App\Models\Product;
    use Livewire\Component;
    use Livewire\WithPagination;
    use Illuminate\Database\Eloquent\Builder;
    class Products extends Component
    {
        protected $queryString = [ 'producto'=>['except'=>'']];
        public $product = '';
        public $showEditModal = false;
        public $description = '';
        public Product $showing;

        public function mount($product='')
        {
            $this->fill(request()->only('product'));
        }

        protected $listeners = ['showModal'];

        public function showModal($product_id)
        {
            $showing = Product::find($product_id);
            $this->description = $showing->description;
            $this->showEditModal = true;
        }
        public function render()
        {
            return view('livewire.products',[ 'products'=>Product::all() ])->layout('layouts.frontpage');
        }
    }

This is the view 

       <div>
        <x-slot name="header">
            <x-front.navbar />
        </x-slot>
        <div class="lg:container mx-auto flex">
            <div class="md:order-2 w-full md:w-3/4 p-4 md:p-0">
                <div wire:loading.class='opacity-50' class="grid grid-cols-2 gap-4 md:grid-cols-3 products-list ">
                    @forelse ($products as $product)
                    <x-front.product-listed  name="{{ $product->name }}"  wire:click='showModal( {{ $product->id }} )'  :key="'product-listed-'.$product->id"  />
                    @empty
                    <div class="col-span-2 md:col-span-3"> Nothing here </div>
                    @endforelse
                </div>
                <div class="mt-8 w-full">{{ $products->links() }}</div>
            </div>
        </div>
        <x-modal.product wire:model.defer="showEditModal">
            <x-slot name="title">{{ $texto }}</x-slot>
            <x-slot name="description">
                {{ $description }}
            </x-slot>
        </x-modal.product>
    </div>
    
I don't understand how to fire the function that opens the modal (showModal) with the product.
I've tried this with init and emit:

    @if ($product)
    <script>
       Livewire.init('showModal',$product);
    </script>
    @endif

Of course, this doesn't work. 
I can't wrap my head around this and may be I'm confusing the concepts. I'm a newbie with Livewire and Laravel in general. I've tried many things but probably some of them are totally misconceptions because I'm note native english speaker. Can you help me, please? Thanks!

**Comment by ZerglingGo** [comment]
How about using `mount()`?

`mount()` works only once after the component is created.
Example:
```
public function mount()
{
     $this->showModal($this->product);
}
```

**Reply by slzrdev** [reply]
Thanks @ZerglingGo , I tried what you suggested but it doesn't work. 
I'd updated the question to make it clearer. Hope it helps.

I tried as you mentioned and with an if condition, because not always will be a product in the url to the view, and nothing happens.

`public function mount()
    {
        $this->fill(request()->only('search', 'cat', 'perPage', 'product'));
        if($producto){
            $this->showModal($producto);
        }
    }`

**Reply by ZerglingGo** [reply]
Did you try to change `$producto` to `$this->producto` and declare `$producto` property?
